internal/mode/static/nginx/config/upstreams.go

```diff
@@ -35,6 +38,60 @@ func (g GeneratorImpl) executeUpstreams(conf dataplane.Configuration) []executeR
 	return []executeResult{result}
 }
 
+func (g GeneratorImpl) executeStreamUpstreams(conf dataplane.Configuration) []executeResult {
+	upstreams := g.createStreamUpstreams(conf.StreamUpstreams)
+
+	result := executeResult{
+		dest: streamConfigFile,
+		data: helpers.MustExecuteTemplate(upstreamsTemplate, upstreams),
+	}
+
+	return []executeResult{result}
+}
+
+func (g GeneratorImpl) createStreamUpstreams(upstreams []dataplane.Upstream) []stream.Upstream {
+	// capacity is the number of upstreams + 1 for the invalid backend ref upstream
+	ups := make([]stream.Upstream, 0, len(upstreams)+1)
+
+	for _, u := range upstreams {
+		ups = append(ups, g.createStreamUpstream(u))
+	}
+
+	return ups
+}
+
+func (g GeneratorImpl) createStreamUpstream(up dataplane.Upstream) stream.Upstream {
+	zoneSize := ossZoneSize
+	if g.plus {
+		zoneSize = plusZoneSize
```

Comment:
Looks like this needs to be covered by a test.

Reply:
ok, I thought that it wasn't necessary since its the same as http upstream code, but I suppose I can just copy paste the same tests for stream upstreams.

Reply:
It's the same as the http upstream code now, but that could change at any moment.